src/main/java/com/spotify/github/v3/repos/requests/FileCreate.java

```diff
@@ -0,0 +1,49 @@
+/*-
+ * -\-\-
+ * github-api
+ * --
+ * Copyright (C) 2016 - 2023 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.github.v3.repos.requests;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.spotify.github.GithubStyle;
+import org.immutables.value.Value;
+
+import javax.annotation.Nullable;
+
+/**
+ * Request to create a file.
+ */
+@Value.Immutable
+@GithubStyle
+@JsonSerialize(as = ImmutableFileCreate.class)
+@JsonDeserialize(as = ImmutableFileCreate.class)
+public interface FileCreate {
```

Comment:
Went with request object, not sure where do you draw the line for sensible amount of parameters. Was on the fence with 3 here, with 4 for `FileUpdate` it seemed more clear cut.